Source/WebCore/platform/ScrollSnapAnimatorState.cpp

```diff
@@ -166,56 +182,54 @@ bool ScrollSnapAnimatorState::resnapAfterLayout(ScrollOffset scrollOffset, const
     auto snapOffsetsVertical = snapOffsetsForAxis(ScrollEventAxis::Vertical);
     auto snapOffsetsHorizontal = snapOffsetsForAxis(ScrollEventAxis::Horizontal);
     
+    auto previouslySnappedBoxes = std::exchange(m_currentlySnappedBoxes, { });
```

Comment:
The way I viewed m_currentlySnappedBoxes is that when this function is run, currentlySnappedOffsetsForAxis are not really the currently snapped offsets, but rather candidate offsets that should be chosen based off of m_currentlySnappedBoxes. So I don’t think its right to call m_currentlySnappedBoxes previouslySnappedBoxes. I think it only makes sense to change m_currentlySnappedBoxes once ScrollSnapAnimatorState::resnapAfterLayout completes.